### PR TITLE
feat[CL]: separate fees into different module account

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #4786 

## What is the purpose of the change

We currently keep fees generated in the same module account that we keep pool funds. This increases the upper bound of loss in the event a bug is introduced. This PR introduces as separate module account that is strictly used to contain and distribute fees.

The key changes are:

- Pool message in the proto file now contains a `fees_address` field that holds the address for fees from swaps​
- Both swap in and out messages now return how much of the swap in was fees
- The `updatePoolForSwap` function has been updated to include `totalFees` as an argument
- In `fees.go`, the `SendCoins` function has been updated to send the fees to `pool.GetFeesAddress()` instead of `pool.GetAddress()`

## Testing and Verifying

Existing swap and fee tests are modified to check for fee value separately.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/concentrated-liquidity/swaps_test.go [hunk 1]
```diff
@@ -3017,7 +3043,7 @@ func (s *KeeperTestSuite) TestFunctionalSwaps() {
 	expectedSqrtPrice := osmomath.MustNewDecFromStr("71.74138432587113364823838192")
 	actualSqrtPrice := osmomath.BigDecFromSDKDec(clPool.GetCurrentSqrtPrice())
 	expectedTokenIn := swapCoin1.Amount.Mul(sdk.NewInt(int64(positions.numSwaps)))
-	expectedTokenOut := sdk.NewInt(983645)
+	expectedTokenOut := sdk.NewInt(982676)
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/swaps_test.go:3046]: All of these new numbers are the correct numbers that show in the python code in the comments. Someone changed them in order to use .02% fee but I changed this back to .03% fee like the python code shows

File: x/concentrated-liquidity/fees_test.go [hunk 2]
```diff
@@ -923,7 +923,7 @@ func (s *KeeperTestSuite) TestQueryAndCollectFees() {
 			err = clKeeper.ChargeFee(ctx, validPoolId, tc.globalFeeGrowth[0])
 			s.Require().NoError(err)
 
-			poolBalanceBeforeCollect := s.App.BankKeeper.GetBalance(ctx, validPool.GetAddress(), ETH)
+			poolBalanceBeforeCollect := s.App.BankKeeper.GetBalance(ctx, validPool.GetFeesAddress(), ETH)
```
Comment (mattverse) [inline x/concentrated-liquidity/fees_test.go:926]:
```suggestion
			poolFeeBalanceBeforeCollect := s.App.BankKeeper.GetBalance(ctx, validPool.GetFeesAddress(), ETH)
```

File: x/concentrated-liquidity/fees_test.go [hunk 3]
```diff
@@ -949,7 +949,7 @@ func (s *KeeperTestSuite) TestQueryAndCollectFees() {
 
 			// Assertions.
 
-			poolBalanceAfterCollect := s.App.BankKeeper.GetBalance(ctx, validPool.GetAddress(), ETH)
+			poolBalanceAfterCollect := s.App.BankKeeper.GetBalance(ctx, validPool.GetFeesAddress(), ETH)
```
Comment (mattverse) [inline x/concentrated-liquidity/fees_test.go:952]:
```suggestion
			poolFeeBalanceAfterCollect := s.App.BankKeeper.GetBalance(ctx, validPool.GetFeesAddress(), ETH)
```

**Review by mattverse — APPROVED** [review]
This LGTM

File: x/concentrated-liquidity/swaps.go [hunk 4]
```diff
@@ -390,20 +391,21 @@ func (k Keeper) computeOutAmtGivenIn(
 		// We deduct the amount of tokens we input in the computeSwapStep above from the user's defined tokenIn amount
 		swapState.amountSpecifiedRemaining.SubMut(amountIn.Add(feeCharge))
 		// We add the amount of tokens we received (amountOut) from the computeSwapStep above to the amountCalculated accumulator
-		swapState.amountCalculated.AddMut(amountOut)
+		swapState.amountCalculated = swapState.amountCalculated.AddMut(amountOut)
```
[the remark this answers is not in the document]
Comment (czarcas7ic) [reply x/concentrated-liquidity/swaps.go]: Yeah you are right, I don't remember making this change. Must have been a branch conflict, thanks for fixing :)

**Review by p0mvn — APPROVED** [review]
🥪

**Pull request merged**